src/main/java/org/example/controllers/AuthController.java

```diff
@@ -0,0 +1,53 @@
+package org.example.controllers;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiKeyAuthDefinition;
+import io.swagger.annotations.SecurityDefinition;
+import io.swagger.annotations.SwaggerDefinition;
+import org.example.exceptions.InvalidException;
+import org.example.models.LoginRequest;
+import org.example.services.AuthService;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.sql.SQLException;
+
+@Api("Auth API")
+@Path("/api/auth")
+@SwaggerDefinition(
+        securityDefinition = @SecurityDefinition(
+                apiKeyAuthDefinitions = {
+                        @ApiKeyAuthDefinition(
+                                key = HttpHeaders.AUTHORIZATION,
+                                name = HttpHeaders.AUTHORIZATION,
+                                in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER
+                        )
+                }
+        )
+)
+public class AuthController {
+    AuthService authService;
+
+    public AuthController(final AuthService authService) {
+        this.authService = authService;
+    }
+
+    @POST
+    @Path("/login")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response login(final LoginRequest loginRequest) {
```

Comment:
You are missing validation cause there are restraints on your DB for the username/password fields, so you should check if any of those are exceeded before going to the DB. The same checks should also be done on the front end so you aren't making pointless API calls